### PR TITLE
Windows nodes support in GCE cloud provider

### PR DESCRIPTION
**Pull request by losipiuk**

### CODE REVIEW

File: cluster-autoscaler/cloudprovider/gce/templates_test.go [hunk 1]
```diff
@@ -276,35 +276,46 @@ func TestGetAcceleratorCount(t *testing.T) {
 
 func TestBuildCapacityMemory(t *testing.T) {
 	type testCase struct {
-		physicalMemory int64
-		capacityMemory int64
-		physicalCpu    int64
+		physicalCpu            int64
+		physicalMemory         int64
+		os                     OperatingSystem
+		expectecCapacityMemory int64
```
Comment (aleksandra-malinowska) [inline cluster-autoscaler/cloudprovider/gce/templates_test.go]: s/expectec/expected

File: cluster-autoscaler/cloudprovider/gce/templates.go [hunk 2]
```diff
@@ -110,56 +125,57 @@ func (t *GceTemplateBuilder) BuildNodeFromTemplate(mig Mig, template *gce.Instan
...
 	node.ObjectMeta = metav1.ObjectMeta{
 		Name:     nodeName,
 		SelfLink: fmt.Sprintf("/api/v1/nodes/%s", nodeName),
 		Labels:   map[string]string{},
 	}
 
-	capacity, err := t.BuildCapacity(cpu, mem, template.Properties.GuestAccelerators)
+	os := extractOperatingSystemFromKubeEnv(kubeEnvValue)
+	if os == OperatingSystemUnknown {
+		os = DefaultOperatingSystem
```
Comment (aleksandra-malinowska) [inline cluster-autoscaler/cloudprovider/gce/templates.go]: nit: why define DefaultOperatingSystem near extractOperatingSystemFromKubeEnv, but only use it here?

**Comment by aleksandra-malinowska** [comment]
/lgtm
/approve

/hold 
Just a couple of nits, feel free to remove hold if you prefer

File: cluster-autoscaler/cloudprovider/gce/templates.go (hunk 2 above)
Comment (losipiuk) [reply cluster-autoscaler/cloudprovider/gce/templates.go]: Idk. I guess I defined it here just randomly, by the type definition. 
On the second though I will change the `extractOperatingSystemFromKubeEnv` to return DefaultOperatingSystem if no is defined in `AUTOSCALER_ENV_VARS` and only return `OperatingSystemUnknown` if something actually unknown is there. Then I will raise an error if `OperatingSystemUnknown` is returned

File: cluster-autoscaler/cloudprovider/gce/templates_test.go [hunk 3]
```diff
@@ -127,23 +127,52 @@ func TestBuildNodeFromTemplateSetsResources(t *testing.T) {
...
+			os:              OperatingSystemLinux,
+			expectedOsLabel: "linux",
+		},
+		{
+			name:            "os windows",
+			os:              OperatingSystemWindows,
+			expectedOsLabel: "windows",
+		},
+		{
+			name:            "os unknown",
```
Comment (aleksandra-malinowska) [inline cluster-autoscaler/cloudprovider/gce/templates_test.go]: test fails

**Pull request merged**